ex18_praticepython.py

```diff
@@ -1,54 +1,39 @@
-#ex18
-#import random
+#ex18   #rules: https://en.wikipedia.org/wiki/Bulls_and_Cows
+import random
 
 def compare(lottery,guess):
-    #first cow (guess in the correct place ) bull (in wrong place)
-    cowbull=[0,0]
+    cowbull=[0,0]   #index zero = bull, index 1 = cow
     
-    for i in range(4):
-      for j in range(4):
-        if guess[i] == lottery[j]:
+    for i in range(len(lottery)):
+      for j in range(len(lottery)):
+        if int(guess[i]) == lottery[j]:
           if i == j:
               cowbull[0]+=1
-          elif guess[j] == lottery[j]:
-              cowbull[1]=cowbull[1]      
-          else:
+              guess[i]=10
+          if i != j:
               cowbull[1]+=1
-#    for i,lottery_val in enumerate(lottery):
-#        #print (i,lottery_val)
-#        for j,guess_val in enumerate(guess):
-#            
-#            if lottery_val==guess_val:
-#                if i==j:
-#                    cowbull[0]+=1
-#                else:
-#                    cowbull[1]+=1
-            
     return cowbull       
 
 if __name__=="__main__":
     playing = True #gotta play the game
-    #lottery=[]
-    #for i in range(4):
-    #    i=random.randint(0,9)
-    #    lottery.append(i)    
+    lottery=random.sample(range(0,9),4)  #random.sample gives me unique digits 
     score = 0
-    lottery2=[6, 1, 6, 8]
 
     while playing:
-        guess2 = input("Give me your best guess!")
-        if guess2 == "exit":
+        guess = list(input("Give me your best guess! The digits must be all different. If the matching digits are in their right positions, they are bulls, if in different positions, they are cows."))
+        guess = [int(i) for i in guess]

```

Comment:
Brakuje ostatniego elementu żeby program działał poprawnie.
Skoro już wymagasz unikalnych cyfr to należy to sprawdzić i nie pozwolić użytkownikowi na oszustwo przez wpisywanie wszystkich cyfr jednakowych (łatwiej zgadnąć która cyfra jest krową).

Reply:
OK